core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -257,8 +292,9 @@ private static Map<String, String> tokenExchangeRequest(
       formData.put(ACTOR_TOKEN, actorToken);
       formData.put(ACTOR_TOKEN_TYPE, actorTokenType);
     }
+    formData.putAll(optionalParams);
 
-    return formData.build();
+    return formData.buildKeepingLast();
```

Comment:
please note, changed it to `buildKeepingLast()` , I noticed everywhere else `build` is used which throws exception if it encounters duplicate keys.